src/MainnetController.sol

```diff
@@ -180,6 +195,31 @@ contract MainnetController is AccessControl {
         emit MaxSlippageSet(pool, maxSlippage);
     }
 
+    function setUniswapV3Router(address router) external {
+        _checkRole(DEFAULT_ADMIN_ROLE);
+        uniswapV3Router = router;
+        emit UniswapV3RouterSet(router);
+    }
+
+    function setUniswapV3PositionManager(address positionManager) external {
+        _checkRole(DEFAULT_ADMIN_ROLE);
+        uniswapV3PositionManager = positionManager;
+        emit UniswapV3PositionManagerSet(positionManager);
+    }
+
+    function setUniswapV3PoolParams(address pool, UniswapV3Lib.UniswapV3PoolParams memory params) external {
+        _checkRole(DEFAULT_ADMIN_ROLE);
+
+        require(
+            params.swapMaxTickDelta > 0 &&
+            params.swapMaxTickDelta <= UniswapV3Lib.MAX_TICK_DELTA,
+            "MainnetController/max-tick-delta-out-of-bounds"
+        );
+
+        uniswapV3PoolParams[pool] = params;
+        emit UniswapV3PoolParamsSet(pool, params);
+    }
```

Comment:
It would be better to separate the admin functions into two separate calls: One for the tick, one for the twap seconds
Rather than passing in a struct like this, I don't think that we generally do that for proxy governance functions. Otherwise, we always have to reset one parameter when we're setting the other parameter. We may not want them always set in the same spell. This creates a potential for resetting it to zero by accident.

Reply:
good point! split this into two